core/src/main/java/nl/nn/adapterframework/doc/CumulativeGroupCreator.java

```diff
@@ -0,0 +1,119 @@
+package nl.nn.adapterframework.doc;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import nl.nn.adapterframework.doc.model.FrankElement;
+
+/**
+ * Strategy class for handling overrides while writing an XML schema from a FrankDocModel.
+ * This class is reused for building attribute groups and building config child groups.
+ * The use of this class is explained for attribute groups.
+ *
+ * For each FrankElement in the model, we need an XSD attributeGroup that lists
+ * all attributes and all inherited attributes. To keep the XSD small, we would like
+ * to list the inherited attributes by referencing the attributeGroup of the parent.
+ * This is not possible however when the current FrankElement overrides attributes.
+ * In this case, the data from the current FrankElement is needed and attribute that
```

Comment:
This text is not present anymore, because we have base class ElementChild and callback interface CumulativeChildHandler. I hope the code explains itself now.

Reply:
De formuleringen zijn nu anders. De uitleg staat nu in DocWriterNew en daar heb ik voorbeelden toegevoegd om te laten zien wat we willen bereiken.